### PR TITLE
USAGOV-106 nav-items-menu-block

### PR DESCRIPTION
**Pull request by cwacht**
## Jira Task

https://cm-jira.usa.gov/browse/USAGOV-106

## Description

move pagetype logic out of twig template

## Checklist for the Developer

- [ ] A link to the JIRA ticket has been added above.
- [ ] The JIRA ticket identifies the desired result of this change.
- [ ] The JIRA ticket contains clear acceptance criteria.
- [ ] The JIRA ticket contains clear testing steps.
- [ ] The JIRA ticket contains a description of "Done".
- [ ] Any preparation/installation/update steps required for this code to work properly (drush commands, scripts, configuration updates) are documented in the ticket.

## Checklist for the Peer Reviewers
- [ ] The branch name of this PR matches the project standards.
- [ ] QA steps are followed and any changes to process are updated in the ticket.
- [ ] Code Standards are followed, there are no bad practices in use.
- [ ] Config changes (if any) include only necessary changes.
- [ ] No errors in Drupal or Client Browser.
- [ ] Code has been tested locally (if possible).
- [ ] Following AC and Testing Steps verify changes work as expected.
- [ ] There are no known side-effects outside of expected behavior.

### CODE REVIEW

File: web/themes/custom/usagov/usagov.theme [hunk 1]
```diff
@@ -36,47 +37,53 @@ function children_of_active_menu_item(&$items) {
  * Process block content
  */
 function usagov_preprocess_block(&$variables) {
-  $variables['node'] = \Drupal::routeMatch()->getParameter('node');
+  $node = \Drupal::routeMatch()->getParameter('node');
+  $variables['node'] = $node;
+
+  if ($node->hasField('field_page_type') && $node->get('field_page_type')[0]->target_id) {
```
Comment (akf) [inline web/themes/custom/usagov/usagov.theme]: Okay, this is where I'm hitting an error message when I try to log in to the cms: 

Error: Call to a member function hasField() on null in /var/www/web/themes/custom/usagov/usagov.theme

In this context, we need to account for there not being a node. The old arrangement sort of did that by accident.

**Review by akf — CHANGES_REQUESTED** [review]
Just formalizing -- this needs some edits before it can be merged. Currently it breaks login to the cms. We'll also want to remove the {{ dd() }} debugging and circleci config changes, of course.

File: web/themes/custom/usagov/usagov.theme (hunk 1 above)
Comment (cwacht) [reply web/themes/custom/usagov/usagov.theme]: Thanks! I've added the check for $node and removed the debugging code

**Review by akf — APPROVED** [review]
Looks good now.

**Pull request merged**